src/popper/runner_host.py

```diff
@@ -293,6 +293,212 @@ def _update_with_engine_config(self, container_args):
+        ]
+        for k, v in update_with.get("environment", dict()).items():
+            container_args["environment"].update({k: v})
+
+        for k, v in update_with.items():
+            if k not in container_args.keys():
+                container_args[k] = update_with[k]
+
+    def _create_container(self, cid, step):
+        build, img, tag, build_ctx_path = self._get_build_info(step)
+
+        if build:
+            log.info(f"[{step.id}] podman build {img}:{tag} {build_ctx_path}")
+            if not self._config.dry_run:
+                pass
+        elif not self._config.skip_pull and not step.skip_pull:
+            log.info(f"[{step.id}] podman build {img}:{tag}")
+            if not self._config.dry_run:
+                cmd = ["podman", "pull", f"{img}:{tag}"]
+                HostRunner._exec_cmd(cmd, logging=False)
+
+        if self._config.dry_run:
+            return
+
+        container_args = self._get_container_kwargs(step, f"{img}:{tag}", cid)
+
+        log.debug(f"Container args: {container_args}")
+
+        msg = f"[{step.id}] podman create name={cid}"
+        msg += f' image={container_args["image"]}'
+        if container_args["entrypoint"]:
+            msg += f' entrypoint={container_args["entrypoint"]}'
+        if container_args["command"]:
+            msg += f' command={container_args["command"]}'
+        log.info(msg)
+
+        cmd = ["podman", "create"]
+        if "command" in container_args and container_args["command"]:
+            cmd.append(container_args["command"])
+        if "name" in container_args and container_args["name"]:
+            cmd.extend(["--name", container_args["name"]])
+        if "volumes" in container_args and container_args["volumes"]:
+            cmd.extend(["--volume", container_args["volumes"][0]])
+        if "environment" in container_args and container_args["environment"]:
+            for i, j in container_args["environment"].items():
+                cmd.extend(["--env", f"{i}={j}"])
+        if "entrypoint" in container_args and container_args["entrypoint"]:
+            cmd.extend(["--entrypoint", container_args["entrypoint"][0]])
+        if "detach" in container_args and container_args["detach"] == True:
+            cmd.append("-d")
+        if "working_dir" in container_args and container_args["working_dir"]:
+            cmd.extend(["-w", container_args["working_dir"]])
+        if "hostname" in container_args and container_args["hostname"]:
+            cmd.extend(["-h", container_args["hostname"]])
+        if "tty" in container_args and container_args["tty"]:
+            cmd.extend(["-t", container_args["tty"]])
+        if "domainname" in container_args and container_args["domainname"]:
+            cmd.extend(["--domainname", container_args["domainname"]])
+        if "image" in container_args and container_args["image"]:
+            cmd.append(container_args["image"])
+
+        _, _, container = HostRunner._exec_cmd(cmd, logging=False)
```

Comment:
we need to check for failure, as `HostRunner._exec_cmd` doesn't throw an exception if the command failed

Reply:
Oh, including a try and except in here would fix that. I felt there was a way to handle the failure of a workflow from previous functions in the codebase. I will try to make changes however.

Reply:
Yeah I found [this](https://github.com/getpopper/popper/blob/250d69e7627808842843cb2316c33874e078b7eb/src/popper/runner_host.py#L93), I thought it would cover exceptions.

Reply:
Failure here can be checked by looking at the value of the exit code returned by the `_exec_cmd` function. If it's non-zero, then something failed

Reply:
I did something like this when refactoring `find_container` method. I should've updated my comments here. Thanks